### ISSUE TITLE
new rule - unused top level private variable

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#7539]
if a symbol is prefixed with a `_` and is not used in the same file, it should be reasonable to assume it's not being used at all
```py
_foo = 1
def _bar(): ...
```

**Comment by DetachHead** [comment]
should also catch unused members of private classes, and unused private members of public classes:

```py
class _Foo:
    asdf = 1 # error

_Foo()
```
```py
class Foo:
    _asdf = 2 # error
```

**Comment by KotlinIsland** [comment]
`_a` means both private and that it is unused. epic!

**Comment by DetachHead** [comment]
maybe `_` means intentionally unused for locally scoped symbols, but private for globally scoped symbols

the rule could also catch these scenarios:
```py
class _Foo: # no error because it's used
    a: int # error because it's unused and in a private class
_Foo()

class Foo:
    _a: int # error because it's private and unused
Foo()

def foo(_bar: int): # no error, local scope so it's intentionally unused
    _baz: int # no error, local scope
    qux: int # error: unused (already handled by F842)
```
